gnovm/pkg/gnolang/alloc.go

```diff
@@ -342,92 +373,168 @@ func (alloc *Allocator) NewHeapItem(tv TypedValue) *HeapItemValue {
 }
 
 // -----------------------------------------------
+// Utilities for obtaining shallow size
 
-func (pv *PackageValue) GetShallowSize() int64 {
-	var (
-		allocFNames     int64
-		allocFBlocks    int64
-		allocFBlocksMap int64
-	)
-
-	for _, name := range pv.FNames {
-		allocFNames += int64(len(name)) + _allocName // string is counted as shallow size.
+// keep sync with copyValueWithRefs() while withRef is true
+func (pv *PackageValue) GetShallowSize(withRef bool) int64 {
+	// .uverse is preloaded
+	if pv.PkgPath == ".uverse" {
+		return 0
 	}
 
-	allocFBlocks = int64(len(pv.FBlocks)) * _allocValue
+	var ss int64
+	ss = allocPackage
 
-	for name := range pv.fBlocksMap {
-		allocFBlocksMap += int64(len(name)) + _allocName // key
-		allocFBlocksMap += _allocPointer                 // *Block
+	if !withRef {
+		return ss
 	}
 
-	return allocPackage + allocFNames + allocFBlocks + allocFBlocksMap
```

Comment:
these were not explicitly allocated, so not counting here. allocate them as well?